package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,331 @@
+This is a numerical example of a known theoretical result that the MSD of a random walk is linear with respect to lagtime, with a slope of :math:`2d`.
+In this expression :math:`d` is the dimensionality of the MSD which for our 3D MSD is equal to 3.
+For comparison we have plotted the line :math:`y=6\tau` to which an ensemble of 3D random walks should converge.
+
+.. _figure-msd: 
+  
+.. figure:: /images/msd_demo_plot.png 
+    :scale: 100 % 
+    :alt: MSD plot 
+
+Note that a segment of the MSD is required to be linear to accurately determine self-diffusivity.
+This linear segment represents the so called "middle" of the MSD plot, where ballistic trajectories at short time-lags are excluded along with poorly averaged data at long time-lags.
+We can select the "middle" of the MSD by indexing the MSD and the time-lags. Appropriately linear segments of the MSD can be confirmed with a log-log plot as is often reccomended [Maginn2019]_ where the "middle" segment can be identified as having a slope of 1.
+
+    >>> plt.loglog(lagtimes, msd)
+    >>> plt.show()
+
+Now that we have identified what segment of our MSD to analyse, lets compute a self-diffusivity.
+
+Computing Self-Diffusivity
+--------------------------------
+Self-diffusivity is closely related to the MSD.
+
+.. math::
+
+   D_d = \frac{1}{2d} \lim_{t \to \infty} \frac{d}{dt} MSD(r_{d}) 
+
+From the MSD, self-diffusivities :math:`D` with the desired dimensionality :math:`d` can be computed by fitting the MSD with respect to the lag time to a linear model. 
+An example of this is shown below, using the MSD computed in the example above. The segment between :math:`\tau = 20` and :math:`\tau = 60` is used to demonstrate selection of an MSD segment.
+
+    >>> from scipy.stats import linregress as lr
+    >>> start_time = 20
+    >>> start_index = int(start_time/timestep)
+    >>> end_time = 60
+    >>> end_index = int(end_time/timestep)
+    >>> linear_model = lr(lagtimes[start_index:end_index], msd[start_index:end_index])
+    >>> slope = linear_model.slope
+    >>> error = linear_model.rvalue
+    >>> D = slope * 1/(2*MSD.dim_fac) #dim_fac is 3 as we computed a 3D msd ('xyz')
+
+We have now computed a self-diffusivity!
+
+
+Notes
+_____
+
+There are several factors that must be taken into account when setting up and processing trajectories for computation of self-diffusivities.
+These include specific instructions around simulation settings, using unwrapped trajectories and maintaining relatively small elapsed time between saved frames. 
+Additionally corrections for finite size effects are sometimes employed along with varied means of estimating errors.
+The reader is directed to the following review, which describes many of the common pitfalls [Maginn2019]_. There are other ways to compute self-diffusivity including from a Green-Kubo integral. At this point in time these methods are beyond the scope of this module
```

Comment:
Is it worthwhile to also mention the recent

von Bülow, S., Bullerjahn, J.T., and Hummer, G. (2020). Systematic errors in diffusion coefficients from long-time molecular dynamics simulations at constant pressure. [arXiv:2003.09205](http://arxiv.org/abs/2003.09205) [Cond-Mat, Physics:Physics].

while you're pointing the reader to other papers?

Reply:
And there is also

I.-C. Yeh and G. Hummer. System-size dependence of diffusion coefficients and viscosities from molecular dynamics simulations with periodic boundary conditions. The Journal of Physical Chemistry B, 108(40):15873–15879, 2004. doi: [10.1021/jp0477147](https://doi.org/10.1021/jp0477147)

Perhaps an addition (new PR) could be a calculation of the finite-size corrected diffusion coefficient?